third_party/compilerworks/README.md

```diff
@@ -0,0 +1,36 @@
+# Using Compilerworks UDFs in BigQuery
+
+## Before You Start
+
+In order to create and use the UDFs within your BigQuery workstreams, ensure that the correct IAM permissions are set.
+
+
+* For creators of the UDFs ensure that they have the following permissions in the dataset of creation:
+  * bigquery.routines.create
+  * bigquery.jobs.create
+
+* For query users of the UDFs please ensure that they have read access to the dataset where the udfs are stored along with the dataset with the actual data.
+  * bigquery.routines.get
+  * bigquery.jobs.create
+  * bigquery.tables.getData
+
+## Creating the UDFs
+
+If you want to create and use the UDFs from this repository in your BigQuery dataset,
+you must provide your dataset name in the following `bqutil` command:
+
+`bq query --project_id YOUR_PROJECT_ID --dataset_id YOUR_DATASET_ID
+--nouse_legacy_sql < cw_stdlib.sql`
```

Comment:
This won't work since all the UDFs in the cw_stdlib.sql have the dataset cw_udf explicitly defined in the DDL. 
I'm ok with removing the dataset from all UDFs and having the user pass in their own via the --dataset_id

Reply:
Removed `cw_udf` from dataset